gdal/frmts/hdf5/bagdataset.cpp

```diff
@@ -3346,8 +3345,7 @@ void BAGDataset::LoadMetadata()
                 {
                     // Found with https://data.ngdc.noaa.gov/platforms/ocean/nos/coast/H12001-H14000/H12525/BAG/H12525_MB_4m_MLLW_1of2.bag
                     // https://github.com/OSGeo/gdal/issues/1643
-                    CPLDebug("BAG", "cornerPoints seems to use a PixelIsArea convention.");
-                    bPixelCenterConvention = false;
+                    CPLError(CE_Failure, CPLE_AppDefined,"Ignoring BAG Upper Right Corner.");
```

Comment:
I'd wish we maintain the ability to handle in a nice way those datasets, even if there are "illegal" given the specification (buggy products do exist in the wild, and when it is possible and reasonable code-wise to handle them, we should try). I'm OK with emitting a message, but rather a CE_Warning than a CE_Failure, and with a more explicit message saying that the product uses a PixelIsArea convention which is non-standard

Reply:
In the final else case with the 3 CPLDebug() messages, you shoud also emit a CPLError(CE_Warning, ...) since that latter case is more annoying that the PixelIsArea case: here the georeferencing is clearly inconsistent !

Reply:
Our understanding of the meaning of these errors is limited and we are more than happy to address these cases as you suggest.  Our only goal here is to ignore the upper right corner rather than scale the resolution.  Since BAGs should not be pixelisarea we wanted to remove that as an option for writing as well.